server/src/main/java/org/cloudfoundry/identity/uaa/security/web/ContentSecurityPolicyFilter.java

```diff
@@ -8,15 +8,27 @@
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 
 public class ContentSecurityPolicyFilter extends OncePerRequestFilter {
+
+    private ContentSecurityPolicyConfiguration configuration = new ContentSecurityPolicyConfiguration();
+    private String cspHeader = null;
+
     @Override
     public void doFilterInternal(HttpServletRequest request,
-                                 HttpServletResponse response, FilterChain chain)
+            HttpServletResponse response, FilterChain chain)
             throws IOException, ServletException {
 
-        response.setHeader("Content-Security-Policy", "script-src 'self'");
+        // If cspHeader were set when the class was instantiated, it would use
+        // the values in `configuration` before they're possibly overwritten
+        // by spring-servlet.xml, which itself reads values from uaa.yml. To
+        // avoid this, we set it lazily here instead.
+        if (this.cspHeader == null) {
```

Comment:
you can also move the setting in the bean constructor, which is the best approach, so then the string can be final